### PR TITLE
Masonry: Add  `handle-item-updates` integration test

### PR DESCRIPTION
**Pull request by dangerismycat**
Another Masonry integration test working! This one tests Masonry's ability to reflow the layout after item sizes have updated. Also fixed other errors related to these integration tests as noted

### CODE REVIEW

File: docs/integration-test-helpers/masonry/MasonryContainer.js [hunk 1]
```diff
@@ -238,9 +238,9 @@ export default class MasonryContainer extends Component<Props, State> {
   };
 
   // $FlowFixMe[unclear-type]
-  renderItem: ({| data: any |}) => Node = ({ data }) => {
+  renderItem: ({| data: any, itemIdx: number |}) => Node = ({ data, itemIdx }) => {
     const { expanded } = this.state;
-    return <ExampleGridItem expanded={expanded} {...data} />;
+    return <ExampleGridItem expanded={expanded} data={data} itemIdx={itemIdx} />;
```
Comment (dangerismycat) [inline docs/integration-test-helpers/masonry/MasonryContainer.js:243]: This wasn't passing through ExampleGridItem's props correctly before — perils of an unsealed Props type!

File: docs/pages/integration-test/masonry.js [hunk 2]
```diff
@@ -7,6 +7,12 @@ import { generateExampleItems } from '../../integration-test-helpers/masonry/ite
 
 const measurementStore = Masonry.createMeasurementStore();
 
+// This is the counterpart to `normalizeValue` in `playwright/masonry/utils/getServerURL.mjs`
+// This will likely need to be updated in the near future
+function booleanize(value: '1' | '0'): boolean {
```
Comment (dangerismycat) [inline docs/pages/integration-test/masonry.js:12]: `normalizeValue` can use more than just `'1' | '0'`for query param values, but I haven't come across that yet — I'll update this when I do

File: playwright/masonry/utils/selectors.mjs [hunk 3]
```diff
@@ -1,10 +1,10 @@
 // @flow strict
 
 const selectors = {
-  afterGrid: '.afterGrid',
-  gridItem: '[data-grid-item], .Collection-Item',
```
Comment (dangerismycat) [inline playwright/masonry/utils/selectors.mjs:5]: This class name isn't valid due to our hashing, but also isn't needed. There aren't other `data-grid-item`s outside of Masonry.

File: playwright/masonry/utils/waitForRenderedItems.mjs [hunk 4]
```diff
@@ -82,7 +82,7 @@ export default async function waitForRenderedItems(
         const { documentElement } = document;
         if (
           typeof scrollHeight !== 'undefined' &&
-          documentElement?.scrollHeight !== scrollHeight
+          documentElement?.scrollHeight >= scrollHeight
```
Comment (dangerismycat) [inline playwright/masonry/utils/waitForRenderedItems.mjs:85]: As previously written, this check is a bit odd — we shouldn't need to care that we're at an exact scrollHeight, just that we've surpassed a minimum

File: playwright/playwright.config.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 // @flow strict
 const { devices } = require('@playwright/test');
 
-const reporter /*: string */ = process.env.CI ? 'github' : 'list';
+const reporter /*: 'github' | 'list' */ = process.env.CI ? 'github' : 'list';
```
Comment (dangerismycat) [inline playwright/playwright.config.js:4]: Does this matter? No. Is it more exact and technically more safe? Sure. ¯\_(ツ)_/¯

**Review by AlbertCarreras — APPROVED** [review]

**Pull request merged**